spec/models/user_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject(:user) { FactoryBot.build(:user)}
+
+  it 'has a valid factory' do
+    expect(user.validate!).to eq(true)
+  end
+
+  it { is_expected.to have_many(:completed_lessons) }
+
+  it { is_expected.to have_many(:lessons).through(:completed_lessons) }
+
+  it { is_expected.to validate_presence_of(:email) }
+
+  it 'invalidates email format with no @' do
+    user.email = 'email.com'
+    expect(user.valid?).to eq(false)
+  end
+
+  it 'invalidates email format with no .' do
```

Comment:
You can group similar tests in a `context` block

Documentation: https://relishapp.com/rspec/rspec-core/v/3-9/docs/example-groups/basic-structure-describe-it#nested-example-groups-(using-%60context%60)

Reply:
This is how I changed the code: https://github.com/TeamBits-2020/ScienceMotions-rails/pull/2/commits/18d02f3549612b3021e35fd6b9bb654b2cc39973